### ISSUE TITLE
"Folder is not writable by user abc"

### ISSUE THREAD

**Issue by tiagoad** [linuxserver/docker-radarr#30]
I mounted a host directory on /usenet and tried to add /usenet/movies as a path to radarr. I always get "Folder is not writable by user abc", even when I set /usenet as owned by abc:abc and change mode to 777. Only this error appears in the logs as well.

If I attach a shell to the running container I can create, read and modify anything in the directory.

<img width="707" alt="image" src="https://user-images.githubusercontent.com/911097/51274383-f53ac800-19c6-11e9-8a37-5244c3a6abe7.png">

**Comment by tobbenb** [comment]
It looks like you set the permissions inside the container. The user you set in PUID/GUID needs read/write permissions on the folder on host that you mapped to /usenet.

**Comment by homerr** [comment]
Can you provide your docker run command, or docker-compose if appropriate please

**Comment by homerr** [comment]
Going to close this as it appears to have gone stale - if you need support with using the container please join us in Discord where available team members and the community can offer you further assistance.

https://discord.gg/YWrKVTn

**Comment by scottocs11** [comment]
For my sonarr container, I changed the PUDI and GUID to 0 instead of 1000.
I'm very new to Docker, so maybe this isn't recommended, but now it has write access to my network shares (and would probably work for your local shares).

**Comment by forsomefun2** [comment]
This is an issue with the container I believe.
The user does have write access. For some reason the container isn't returning valid values for rclone mounts. 
The only way I was able to fix it was to 


Add a fake movie or do the bulk import before starting the mount.
Stop the container
Restart the mount 
Restart the container

That will fix the mounting issues.

**Comment by pixelrogue1** [comment]
"Folder is not writable by user abc"

• MacOS 10.15.4
• Docker Desktop 2.2.0.5 (43884) running Sonarr/Radarr/Jackett

**ATTEMPTS TO FIX**
• Grant Full Disk Access & Folders to bash, sh, smbd,sshd,Terminal, Docker (temporary to find a fix.)
• Full Read/Write permissions to MacOS target folders (applied to enclosed directories)
• Redirected URLS and downloads to less secured public shared space
• Create new directory structures
• Use the same directories for sonarr/radarr/jackett
• sudo chown -R $(whoami) for paths
• Failed attempts at command line fixes from others w/this problem
• Reboots
• Rebuilds of Sonarr/Jacket/Radarr containers
• Delete and reinstall images & containers for Sonarr/Jackett/Radarr
• Delete and reinstall Docker
• Ensured selected directory is read/write to everyone (user, staff, everyone =Read Write)
• Confirm Docker is running from upper level Applications directory (not at User Level)

**Comment by tobbenb** [comment]
This is a closed issue. For support you should use our discord or discourse forum.
We also still don't test containers on Mac, so we don't really support it.

**Comment by DylanAlloy** [comment]
Hey, happy to confirm that this happens on Ubuntu as well.

Also curious as to why you assumed someone changed settings in the Docker instance (lol, I don't know why someone would do that and then come here to complain...). You haven't checked to see what the cause could be clearly. 

Thankful that others did the due diligence. 

Not supporting Mac; okay. How about Ubuntu?

**Comment by aptalca** [comment]
What about ubuntu? This image has 1.2 billion pulls on docker hub. Most of our team members use it daily. If you're having basic permission errors, it's likely pebcak and not a bug. You can use discord or discourse as advised above.

**Comment by auroraflux** [comment]
Using stock Dockstarter settings on a stock Ubuntu 18.04 setup, no changes to any of the mounts in the Sonarr container, and my UID/GID are very much still 1000: 

```
uid=1000(user) gid=1000(user) groups=1000(user),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),108(lxd),999(docker)
```

Seeing a Sonarr and Radarr are almost the same piece of software and I get the same issue, I'm posting this here. 

I still get "Folder is not writable by user abc" any time I try to add the `/tv` or `/data` directory. 

I have my NFS share natively mounted (not through Docker, just standard `mount` command) to `/mnt/medialibrary/tv/` (which, per the stock settings, is translated to `tv` on the container). It's been chmod'd to 777 (recursively). 

Sabdnzbd+ uses the exact same PID/GID and has no issues writing to this directory. 

I'm testing Dockstarter out while I consider switching over from a standard install to a containerized install of all my automation software. I'm not new to using any of these apps, and certainly not new to Sonarr. 

There might be 1.2 billion pulls of the Sonarr image, but all that means is that they are doing something that I'm not doing, because I'm using all the settings stock the way that Dockstarter proposes they be used. 

Anyone have any advice here?

**Comment by auroraflux** [comment]
I've figured it out. Sonarr/Radarr containers don't seem to like to write to host directories that have NFS shares mounted to them through the host system. 

I had to first add an NFS volume (I used Portainer to do this out of laziness), and then manually add that mount to the Sonarr container (mapped the NFS volume I created to `/tv`). 

Don't really know what Sonarr's problem is with this, it's kind of annoying that Sabnzbd doesn't seem to have an issue with using native NFS mounts instead of Docker volume-based NFS mounts, but I guess it works.

**Comment by ch00w00** [comment]
@auroraflux - can I ask how you did this? I tried adding NFS volume in Portainer for my media folder, but this didnt seem to get me any further - all I could see was the mount point folder, rather than it pointing to the media folder. I am very new to this so sorry if this is a dumb thing to ask. Everything I have tried so far still gives me the error about not writable by user abc. thanks

**Comment by Deliganli** [comment]
my docker-compose

```
  radarr:
    image: linuxserver/radarr
    container_name: radarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Istanbul
      - UMASK_SET=022
    volumes:
      - ~/.mediacenter/configs/radarr:/config
      - ~/projects/mediacenter/content:/content
    ports:
      - 7878:7878
    restart: unless-stopped
```

this is the log I managed to retrieve

```

radarr         | [Debug] Api: [GET] /api/filesystem?includeFiles=false&path=%2Fcontent%2Fmedia%2Ffilms: 200.OK (3 ms)
radarr         | [Trace] Http: Req: 158 [GET] /api/filesystem?includeFiles=false&path=%2Fcontent%2Fmedia%2Ffilms%2F
radarr         | [Trace] Http: Res: 158 [GET] /api/filesystem?includeFiles=false&path=%2Fcontent%2Fmedia%2Ffilms%2F: 200.OK (3 ms)
radarr         | [Debug] Api: [GET] /api/filesystem?includeFiles=false&path=%2Fcontent%2Fmedia%2Ffilms%2F: 200.OK (3 ms)
radarr         | [Trace] Http: Req: 159 [POST] /api/rootfolder
radarr         | [Trace] ConfigService: Using default config value for 'downloadedmoviesfolder' defaultValue:''
radarr         | [Trace] DiskProviderBase: Directory '/content/media/films/' isn't writable. Access to the path "/content/media/films/radarr_write_test.txt" is denied.
radarr         | [Trace] NzbDroneErrorPipeline: Handling Exception
radarr         | [Warn] NzbDroneErrorPipeline: Invalid request Validation failed:
radarr         |  -- Folder is not writable by user abc
radarr         | [Trace] Http: Res: 159 [POST] /api/rootfolder: 400.BadRequest (36 ms)
radarr         | [Debug] Api: [POST] /api/rootfolder: 400.BadRequest (36 ms)
```

as you can see below it looks like container has permissions to write to that directory

```

pi@raspberrypi:~/projects/mediacenter $ docker exec -it radarr /bin/bash
root@5653ea5b4b2e:/# cd /content/media/films/
root@5653ea5b4b2e:/content/media/films# ll
total 8
drwxr-xr-x 2 root root 4096 Aug  3 23:48 ./
drwxr-xr-x 5 root root 4096 Aug  3 23:45 ../
root@5653ea5b4b2e:/content/media/films# touch topkek
root@5653ea5b4b2e:/content/media/films# ll
total 8
drwxr-xr-x 2 root root 4096 Aug  4 00:03 ./
drwxr-xr-x 5 root root 4096 Aug  3 23:45 ../
-rw-r--r-- 1 root root    0 Aug  4 00:03 topkek
root@5653ea5b4b2e:/content/media/films# rm topkek
root@5653ea5b4b2e:/content/media/films# ll
total 8
drwxr-xr-x 2 root root 4096 Aug  4 00:03 ./
drwxr-xr-x 5 root root 4096 Aug  3 23:45 ../
root@5653ea5b4b2e:/content/media/films#
```

**Comment by basings** [comment]
Same error `Folder is not writable by user abc` for me with version 5.3.6
Setting `PUID=0 and PGID=0` solves the issue. Took me days. That's why I confirm the solution years later on a closed issue.

````
version: "3"
services:
  radarr:
    image: lscr.io/linuxserver/radarr:latest
    container_name: radarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London
    volumes:
      - ./radarr-config:/config:z
      - ./data:/data:z
    restart: always
    ports:
      - 7878:7878
````

**Comment by Tarnest** [comment]
For anyone who sees this in the future, make sure when referring to specific folders in your compose.yml you use the full path.

Rather than `~/path/to/movies` do `/home/username/path/to/movies`.

**Comment by suquamish** [comment]
I really hate to necropost, but I got here becasue of the error message via Google in August of 2024...

If you find yourself seeing this error message, and like [Deliganli](https://github.com/linuxserver/docker-radarr/issues/30#issuecomment-668264755) everything looks copacetic, you can simply install sql lite on your docker container (`apk add sqlite`) and manually insert the path into your `RootFolders` table.
```
insert into RootFolders (Path) values ('/path/to/movies');
```

This, of course, only helps if you're using a persistent configuration.

**Comment by clebemachado** [comment]
Well, I had created the movie folder with my root user. 
Check your user's PUID and PGID using the id command and check if it is in accordance with what is stated in docker-compose.

Then see if your user (not root) has permission to record in the movie folder defined on the host, otherwise just assign permissions to it. As it is something personal, it immediately assigns all permissions to the user. 
But it's everyone's risk

sudo chmod 777 /opt/movies


```
version: '3'
services:
  radarr:
    image: lscr.io/linuxserver/radarr:latest
    container_name: radarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Brazil
    volumes:
      - /opt/radarr/data:/config
      - /opt/movies:/movies #optional
      - /opt/downloads:/downloads #optional
    ports:
      - 7878:7878
    restart: unless-stopped
```

**Comment by Alfenstein8** [comment]
I figured a solution that works for me. 
when I went into the container, the movies directory was owned by root and in group root. So I just changed it to abc:users. 
Here's what I did:
 1. run `docker exec -it radarr sh` to go intro the radarr container.
 2. check permissions of the movies and downloads directories with `ls -l | grep -E 'movies|downloads'`
 For it to work, it should be:
 ```
 drwxr-xr-x   2 abc  users    40 Jan 15 21:08 downloads
 drwxr-xr-x   2 abc  users    40 Jan 15 21:45 movies
 ```
 3. If any of them has `root root` instead of `abc users`, then change it with `chown abc:users movies` or downloads
 
 Now it should work in radarr
